codice/ddf
# DDF-1883 Updated to Solr 5.5

#### What's this PR do?
DDF-1883 Updated to Solr 5.4
- Removed use of deprecated and removed Solr methods
- Reconstructed the Solr war
- Updated Solr config and schema to latest versions
- Shaded Solr war and Solr factory dependencies
- Removed LuxDB indexing and pre-filtering
- Split Solr factory into external and embedded versions to reduce
  run time dependences to create HTTP Solr clients
- Changed core creation logic of Solr factory to better handle
  creating a core while Solr is starting/restarting
- Improved logging of Solr factory
- Moved Solr config files to application feature files that require
  them
- Made Banana provisioner non-blocking and re-enabled

DDF-1883 Updated to Solr 5.5
- Changed banana-int core to banana due to new restrictions on core
  naming.

#### Who is reviewing it?
@kcwire @brendan-hofmann @jaymcnallie @codice/solr 

#### How should this be tested?
- Ingest records
- View records with Catalog UI
- Create a workspace while logged in and verify workspace is persisted 
- Uninstall external catalog provider and install embedded catalog provider and repeat above steps

#### Any background context you want to provide?
Solr 5 does not provide a war.  The Solr admin web app source code was copied into platform/solr/platform-solr-server-standalone/src/main/webapp.  You can ignore code in that directory for this review.

#### What are the relevant tickets?
https://codice.atlassian.net/browse/DDF-1883

#### Screenshots (if appropriate)

#### Questions:
- Does the documentation need an update? Yes, it was updated.
